go-controller/pkg/metrics/workqueue.go

```diff
@@ -0,0 +1,129 @@
+/*
+Modified from k8s.io/component-base/metrics/prometheus/workqueue/metrics.go
```

Comment:
Could we vendor it instead of copy-pasting it?

Reply:
I originally imported it - `import _ "k8s.io/component-base/metrics/prometheus/workqueue"`
That adds component-base and one transient dependency, which isn't too bad.
The downside is that all metrics are simply `workqueue_*`, which deviates from the pattern that's already established in ovn-k8s.

Implementing it this way we don't have any additional dependencies, and we can prefix the stats with `ovnkube_node` or `ovnkube_master`. It required changing a the original code to use `protheus_golang` directly, and to register the metrics in a function, not on `init`. Enough of the structure of the original was kept that I felt it still warranted putting a link here and preserving the original copyright header.

Reply:
OK, got it. Thanks for explaining that.